editor/gui/editor_file_dialog.cpp

```diff
@@ -206,6 +206,10 @@ void EditorFileDialog::_update_theme_item_cache() {
 
 void EditorFileDialog::_notification(int p_what) {
 	switch (p_what) {
+		case NOTIFICATION_POSTINITIALIZE: {
+			set_translation_domain("godot.editor");
```

Comment:
Why it's not in the constructor?

Reply:
It _was_ in the constructor, but then I got this warning:

> WARNING: Attempting to access theme items too early in EditorFileDialog; prefer NOTIFICATION_POSTINITIALIZE and NOTIFICATION_THEME_CHANGED

Reply:
`EditorFileDialog` updates its icons in `NOTIFICATION_TRANSLATION_CHANGED`, maybe some icons have language dependent variants (?)

Reply:
They don't, looks like a bug. It handles 3 notifications at once, `_update_icons()` should be done only on layout direction and theme changed.

Reply:
`_update_icons()` is indeed locale-dependent. A change in language potentially changes the return value of `is_layout_rtl()`:

https://github.com/godotengine/godot/blob/5ebfc8a6ba1dec27157f48fc49ede54ea6676720/editor/gui/editor_file_dialog.cpp#L1422-L1431

Reply:
But that's what `NOTIFICATION_LAYOUT_DIRECTION_CHANGED` is for.

Reply:
According to the implementation, `NOTIFICATION_LAYOUT_DIRECTION_CHANGED` is only emitted when the control changes between `inherit-parent`, `locale-dependent`, `LTR`, and `RTL`. It's not used when locale changes.